cli/crates/common/src/environment.rs

```diff
@@ -330,6 +334,104 @@ fn find_grafbase_configuration(path: &Path, warnings: &mut Vec<Warning>) -> Opti
     }
 }
 
+#[derive(Clone, Copy, Debug)]
+pub enum NodePackageManager {
+    Npm,
+    Pnpm,
+    Yarn,
+}
+
+impl Default for NodePackageManager {
+    fn default() -> Self {
+        Self::Npm
+    }
+}
+
+impl NodePackageManager {
+    #[must_use]
+    pub fn is_npm(self) -> bool {
+        matches!(self, Self::Npm)
+    }
+}
+
+impl fmt::Display for NodePackageManager {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        match self {
+            Self::Npm => f.write_str("npm"),
+            Self::Pnpm => f.write_str("pnpm"),
+            Self::Yarn => f.write_str("yarn"),
+        }
+    }
+}
+
+impl FromStr for NodePackageManager {
+    type Err = CommonError;
+
+    fn from_str(s: &str) -> Result<Self, Self::Err> {
+        match s {
+            "npm" => Ok(NodePackageManager::Npm),
+            "pnpm" => Ok(NodePackageManager::Pnpm),
+            "yarn" => Ok(NodePackageManager::Yarn),
+            _ => {
+                let inner = io::Error::new(
+                    io::ErrorKind::InvalidData,
+                    format!("Unsupported package manager in package.json: {s}"),
+                );
+
+                Err(CommonError::AccessPackageJson(inner))
+            }
+        }
+    }
+}
+
+pub fn install_dependencies(project_root: &Path) -> Result<(), CommonError> {
+    let command = match which::which("npm") {
+        Ok(command) => command,
+        Err(e) => return Err(CommonError::NpmNotFound(Box::new(e))),
```

Comment:
Why is this using a dyn error rather than a specific type?

Reply:
That's what we currently do since e.g. hints can use the internal error or display it, we could go over them and box them at some point though to make it easier for third parties

Reply:
But ok, let's leave it as is and discuss the rest later